chronos/asyncfutures2.nim

```diff
@@ -60,6 +60,10 @@ type
       closure*: iterator(f: Future[T]): FutureBase {.raises: [Defect, CatchableError, Exception], gcsafe.}
     value: T ## Stored value
 
+  # Future with a tuple of possible exception types
+  # eg FuturEx[void, (ValueError, OSError)]
+  FuturEx*[T, E] = ref object of Future[T]
```

Comment:
This spelling is a bit too clever perhaps. I anticipate that people will make less typos with the simpler `FutureEx`.

Reply:
I'd strive to make it non-public to begin with - once it spreads, we can't get rid of it if it's public - however, in particular for non-raising code, there's a great deal of simplification that can be done to `Future` in general.

Reply:
`RaiseTrackingFuture`?